invenio_accounts/datastore.py

```diff
@@ -118,6 +115,14 @@ def find_domain(self, domain):
             .one_or_none()
         )
 
+    def find_domain_by_id(self, domain_id):
+        """Find a domain by id."""
+        return (
+            Domain.query.filter_by(id=domain_id)
+            .options(joinedload(Domain.category_name))
+            .one_or_none()
+        )
+
```

Comment:
```suggestion
    def find_domain(self, domain_or_id):
        """Find a domain by it's value or ID."""
        if isinstance(domain_or_id, str):
            if domain_or_id.isdigit():
                clause = Domain.id == int(domain_or_id)
            else:
                clause = Domain.domain == domain_or_id
        elif isinstance(domain_or_id, int):
            clause = Domain.id == domain_or_id
        else:
            raise ValueError("...")
        return (
            Domain.query.filter(clause)
            .options(joinedload(Domain.category_name))
            .one_or_none()
        )
```

minor: IMHO this class should try to be as flexible/helpful as possible. Thus, enhancing the existing `find_domain` method to do a bit of inspection to accommodate more use cases is fine and keeps the complexity from spreading to other places.